.changeset/entries/80245c83a4e85ea0b6b482617a7855b49f237e1852eaa5a5e4d273c4b6155367.yaml

```diff
@@ -0,0 +1,7 @@
+type: feat
+module: none
+pull_request: 213
+description: 'Add subspace v3 new methods: `MsgGrantTreasuryAuthorization`, `MsgGrantAllowance`
+  and `MsgRevokeAllowance``'
+backward_compatible: true
+date: 2023-05-30T12:42:45.003172948Z
```

Comment:
@manu0466 From my side, to grant or revoke an allowance via contract, the operator would performs the following steps:
- Check if the grantee has already haven the grant or not by Desmos `UserAllowances` or `GroupAllowances` query, instead of using contract
- Perform the `MsgGrantAllowance` or `MsgRevokeAllowance` via contract

Therefore, I believe there is no case for querying `UserAllowances`, `GroupAllowances` inside contract.

What do you think?

Reply:
@manu0466 After rethinking about queries, I decided to add them since `desmos core` open the `UserAllowances` and `GroupAllowances`.

Reply:
@manu0466 After rethinking about queries, I decided to add them since `desmos core` open the `UserAllowances` and `GroupAllowances`.